x/firewood/sync_db.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package firewood
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/ava-labs/firewood-go-ethhash/ffi"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/maybe"
+
+	xsync "github.com/ava-labs/avalanchego/x/sync"
+)
+
+var _ xsync.DB[*RangeProof, *ChangeProof] = (*syncDB)(nil)
+
+// syncDB wraps a Firewood FFI database to implement the xsync.DB interface.
+type syncDB struct {
+	fw   *ffi.Database
+	lock sync.Mutex // TODO: remove this lock once FFI is thread-safe
+}
+
+func wrapSyncDB(db *ffi.Database) *syncDB {
+	return &syncDB{fw: db}
+}
+
+func (db *syncDB) GetMerkleRoot(context.Context) (ids.ID, error) {
+	root, err := db.fw.Root()
+	if err != nil {
+		return ids.ID{}, err
+	}
+	return ids.ID(root), nil
+}
+
+// GetRangeProofAtRoot returns a range proof for x/sync between [start, end].
```

Comment:
nit: w.r.t the functions on this type - these are documented on the interface so duplicative to document here on the implementation